File: data/compilers_comparison/src/examples/second_example/src/func45.c
#include "second_example.h" 
array_t* func45(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array57;
   if (pCounter > 0) {
      array57 = vars->data[--pCounter];
      array57->refC++;
      DEBUG_COPY(array57->id);
   } else {
      array57 = (array_t*)malloc(sizeof(array_t));
      array57->size = 404;
      array57->refC = 1;
      array57->id = 57;
      array57->data = (unsigned int*)malloc(array57->size*sizeof(unsigned int));
      memset(array57->data, 0, array57->size*sizeof(unsigned int));
      DEBUG_NEW(array57->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array57;
   array_t* array58 = func51(&params0, loopsFactor);
   DEBUG_RETURN(array58->id);
   free(params0.data);
   if(PATH0 & 1) {
      array_t* array61;
      if (pCounter > 0) {
         array61 = vars->data[--pCounter];
         array61->refC++;
         DEBUG_COPY(array61->id);
      } else {
         array61 = (array_t*)malloc(sizeof(array_t));
         array61->size = 917;
         array61->refC = 1;
         array61->id = 61;
         array61->data = (unsigned int*)malloc(array61->size*sizeof(unsigned int));
         memset(array61->data, 0, array61->size*sizeof(unsigned int));
         DEBUG_NEW(array61->id);
      }
      unsigned int loop20 = 0;
      unsigned int loopLimit20 = (rand()%loopsFactor)/4 + 1;
      for(; loop20 < loopLimit20; loop20++) {
         for (int i = 0; i < array61->size; i++) {
            array61->data[i]--;
         }
         for (int i = 0; i < array57->size; i++) {
            if (array57->data[i] == 24) { 
               return array57;
            }
         }
      }
      array61->refC--;
      if(array61->refC == 0) {
         free(array61->data);
         free(array61);
         DEBUG_FREE(array61->id);
      }
   }
   else {
      array_t_param params1;
      params1.size = 2;
      params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
      params1.data[0] = array57;
      params1.data[1] = array58;
      array_t* array62 = func52(&params1, loopsFactor);
      DEBUG_RETURN(array62->id);
      free(params1.data);
      array62->refC--;
      if(array62->refC == 0) {
         free(array62->data);
         free(array62);
         DEBUG_FREE(array62->id);
      }
   }
   array57->refC--;
   if(array57->refC == 0) {
      free(array57->data);
      free(array57);
      DEBUG_FREE(array57->id);
   }
   return array58;
}

File: data/multilang_comparison/program_nim/src/func45.nim
proc func45*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array86: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array86 = vars.data[pCounter]
      inc array86.refC
      debugCopy(array86.id)
   else:
      array86 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array86.size = 197
      array86.refC = 1
      array86.id = 86
      array86.data = cast[ptr UncheckedArray[uint32]](alloc(array86.size * sizeof(uint32)))
      zeroMem(array86.data, array86.size * sizeof(uint32))
      debugNew(array86.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array86
   var array87: ptr ArrayT = func51(addr params0, loopsFactor)
   debugReturn(array87.id)
   dealloc(params0.data)
   if (PATH0 and 1'u64) != 0:
      discard
      var array90: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array90 = vars.data[pCounter]
         inc array90.refC
         debugCopy(array90.id)
      else:
         array90 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array90.size = 746
         array90.refC = 1
         array90.id = 90
         array90.data = cast[ptr UncheckedArray[uint32]](alloc(array90.size * sizeof(uint32)))
         zeroMem(array90.data, array90.size * sizeof(uint32))
         debugNew(array90.id)
      let loopLimit29 = (50) div (4) + 1
      for loop29 in 0 ..< loopLimit29:
         discard
         for i in 0 ..< array90.size:
            dec array90.data[i]
         for i in 0 ..< array87.size:
            if array87.data[i] == 22:
               return array87
      var params1: ArrayTParam
      params1.size = 3
      params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
      params1.data[0] = array86
      params1.data[1] = array87
      params1.data[2] = array90
      var array91: ptr ArrayT = func60(addr params1, getPath(), loopsFactor)
      debugReturn(array91.id)
      dealloc(params1.data)
      dec array91.refC
      if array91.refC == 0:
         dealloc(array91.data)
         dealloc(array91)
         debugFree(array91.id)
      dec array90.refC
      if array90.refC == 0:
         dealloc(array90.data)
         dealloc(array90)
         debugFree(array90.id)
   else:
      discard
      var params1: ArrayTParam
      params1.size = 2
      params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
      params1.data[0] = array86
      params1.data[1] = array87
      var array92: ptr ArrayT = func52(addr params1, loopsFactor)
      debugReturn(array92.id)
      dealloc(params1.data)
      dec array92.refC
      if array92.refC == 0:
         dealloc(array92.data)
         dealloc(array92)
         debugFree(array92.id)
   dec array86.refC
   if array86.refC == 0:
      dealloc(array86.data)
      dealloc(array86)
      debugFree(array86.id)
   return array87

